content/posts/irods-development-update-july-2024.md

```diff
@@ -0,0 +1,273 @@
+
+Users of the iRODS Testing Environment will be happy to know that projects for PostgreSQL 16, MySQL 8.4, and MariaDB 10.6, 10.11, and 11.4 are now available.
+
+The GenQuery python module _(i.e. genquery.py)_ which ships with the Python rule engine plugin is gaining support for GenQuery2. Once complete, the `Query` class will give users the ability to choose the GenQuery parser they'd like to use. With these changes, administrators will have an easy migration path from GenQuery1 to GenQuery2. This is especially helpful in situations where the policy makes heavy use of the `Query` class.
+
+The S3 resource plugin received a couple fixes. The first fix resolves an issue where the server was not honoring the number of threads specified by `iput -N`. The second fix resolves an issue with `irsync` and checksum operations failing.
+
+For the Python iRODS client (PRC), session objects now propagate connection timeout information to the socket layer as originally intended. Support for progress bars has been improved. Users will soon be able to use their favorite progress bar library with the PRC. And last but not least, a question about the recently published ISO standard, ISCC, was presented to the community. If you find that interesting and/or have suggestions, please see [https://github.com/irods/python-irodsclient/issues/573](https://github.com/irods/python-irodsclient/issues/573). The issue also provides a link to an implementation of ISCC for the PRC!
+
+For the Automated Ingest tool, we've adjusted the event handler interface to account for changes introduced in an unreleased update. This is a **breaking** change. Users of the the Automated Ingest tool will need to update their event handlers. We've also added a Docker Compose project which makes it possible to easily test the S3 scanner. 
+
+The iRODS HTTP API has received several updates. The next release of the HTTP API will allow users to modify properties of resources, manage SpecificQuery's, apply expiration timestamps to tickets, and manage zones. Not only that, but administrators will soon be able to track activity better due to the server finally growing the ability to log IP addresses. Finally, the interns are making excellent progress on the client wrappers for the HTTP API. The client wrappers will be covered at the next TRiRODS meeting on August 14th, so keep an eye out for that.
+
+The Zone Management Tool is being updated to use the HTTP API instead of the now archived C++ REST API. This work is being handled by one of interns and is nearly done.
```

Comment:
```suggestion
The Zone Management Tool is being updated to use the HTTP API instead of the now archived C++ REST API. This work is being handled by one of our interns and is nearly done.
```

Reply:
Done.